src/plugin/index-jexl-filter/src/java/org/apache/nutch/indexer/filter/JexlIndexingFilter.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nutch.indexer.filter;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Map.Entry;
+
+import org.apache.commons.jexl2.Expression;
+import org.apache.commons.jexl2.JexlContext;
+import org.apache.commons.jexl2.MapContext;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.nutch.crawl.CrawlDatum;
+import org.apache.nutch.crawl.Inlinks;
+import org.apache.nutch.indexer.IndexingException;
+import org.apache.nutch.indexer.IndexingFilter;
+import org.apache.nutch.indexer.NutchDocument;
+import org.apache.nutch.indexer.NutchField;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.parse.Parse;
+import org.apache.nutch.util.JexlUtil;
+import org.apache.nutch.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * An {@link org.apache.nutch.indexer.IndexingFilter} that allows filtering of
+ * documents based on a JEXL expression.
+ *
+ */
+public class JexlIndexingFilter implements IndexingFilter {
+
+	private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	private Configuration conf;
+	private Expression expr;
+
+	@Override
+	public NutchDocument filter(NutchDocument doc, Parse parse, Text url, CrawlDatum datum, Inlinks inlinks) throws IndexingException {
+	    if (expr != null) {
```

Comment:
For what I see: `JexlUtil` returns null if it receives `null` or if cannot parse the expression:

https://github.com/apache/nutch/blob/b36692a6733e906be6fedbce8607397531666836/src/java/org/apache/nutch/util/JexlUtil.java#L72-L74

So if the expression cannot be parsed all the documents will be indexed because of this check, should we do a more strict check? Because since if you already enable the plugin you probably want your documents filtered (the plugin is not enabled by default).

Reply:
On the other hand, if the indexer is enabled and configured, you probably want your documents indexed, right?
As a rule, I think plugins should not do anything when not configured (other than writing to log), but if there is a strong opposite opinion, I can change this.

Reply:
I partially agree with you, but in this case, the purpose of the plugin is to filter documents, so it has a really _active_ behaviour: if the wrong expression is entered you'll get an error logged but you will end with all the documents (that you wanted to exclude in the first place) in the index. 

I would prefer to get an error that the expression could not be parsed (probably due to a syntax error) of course this would cause the index step to fail. 

We could even go one step further: if the plugin is enabled we could treat the expression as a required parameter because otherwise, the plugin is not doing anything, this is done for instance on the `solr-indexer` plugin (will fail if the required parameters are not specified)

So we will not enable this plugin by default but if you enable the plugin you'll need to provide a valid expression.

Reply:
OK, I will throw a RuntimeException from setConf when null/invalid.

Reply:
👍  the if expression on the `filter` method could also be removed then